### PR TITLE
[CheckableButton, ResourceList] fix checkbox alignment 

### PR DESCRIPTION
**Pull request by gwyneplaine**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/441

### WHAT is this pull request doing?

Before: 

https://github.com/Shopify/polaris/assets/12119389/d321464a-e98c-4e4d-91ed-72127f1f8f24

After:

https://github.com/Shopify/polaris/assets/12119389/e42cb80f-b025-453e-8a33-24804e4d47a4

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by gwyneplaine** [comment]
@sarahill [designs](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Uplift-Components-%5Bgen3%E2%80%93alpha%5D?type=design&node-id=53457-26770&t=dvfc101bnmsGLFRa-4) don't seem to account for the fact that the [`Select` element](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Uplift-Components-%5Bgen3%E2%80%93alpha%5D?type=design&node-id=29402-14771&t=dvfc101bnmsGLFRa-4) used for sorting has a dynamic height (`32px` below md and `36px` above md) 

I've adjusted the padding to account for this in this PR. Below `md` we use `var(--p-space-1_5-experimental)` for 6px top and bottom padding, and `var(--p-space-2)` for 8px top and bottom padding above `md`.

I assume this is satisfactory, but we may also want to adjust the height of the header container to be dynamice `48px` above md and `52px` below md for example.  To keep the padding uniform. 

The fix in this PR is not optimal, as it requires us to hardcode the `min-height` for each container element across **multiple** components. Ideally these hardcoded values should be tokenized (--p-control-height) for example. The solution outlined in [this issue ](https://github.com/Shopify/polaris-summer-editions/issues/299) from @jesstelford  would be ideal.

File: polaris-react/src/components/CheckableButton/CheckableButton.scss [hunk 1]
```diff
@@ -40,13 +40,15 @@
 
 .Checkbox {
   pointer-events: none;
```
Comment (gwyneplaine) [inline polaris-react/src/components/CheckableButton/CheckableButton.scss:42]: This bug was ocurring both with the feature flag disabled and enabled. 
This is a more holistic solution to the problem that also cleans up our css.

**Comment by bernardojoaogarcia** [comment]
@gwyneplaine the figma designs aren't accounting for that dynamic height change. 
Your solution seems good 👍. Like you were saying, I would consider making the header dynamic in order to keep the padding consistent. The only question Sara and I have is would the transition feel smooth when the heights are changing?

**Comment by gwyneplaine** [comment]
@bernardojoaogarcia here's it with just the 48px min-height:
resource-list-static-height


and here's it with a variable height based on breakpoint (52px for small screens, 48px for larger screens):
resource-list-dynamic-height

**Comment by jesstelford** [comment]
After looking at he designs, we noticed the search box _is_ changing height between small -> large screens (`height: 32px;` -> `height: 36px;`), and in that case the padding is _not_ changing. ie; the overall height of the header is what changes.

If we want to be consistent with the search box's behaviour, we can leave padding on the Select, but note that will lead to an overall increase of 16px on small screens (8px for search + 8px for Select). An alternative might be to use the padding-reduction approach on the search box too.

**Review by bernardojoaogarcia — APPROVED** [review]

**Review by jesstelford — APPROVED** [review]

**Pull request merged**